ta4j-core/src/main/java/org/ta4j/core/indicators/supertrend/SuperTrendIndicator.java

```diff
@@ -0,0 +1,84 @@
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package org.ta4j.core.indicators.supertrend;
+
+import org.ta4j.core.Bar;
+import org.ta4j.core.BarSeries;
+import org.ta4j.core.indicators.ATRIndicator;
+import org.ta4j.core.indicators.RecursiveCachedIndicator;
+import org.ta4j.core.num.Num;
+
+public class SuperTrendIndicator extends RecursiveCachedIndicator<Num> {
+
+    private final Num ZERO = zero();
+    private final SuperTrendUpperBandIndicator superTrendUpperBandIndicator;
+    private final SuperTrendLowerBandIndicator superTrendLowerBandIndicator;
+
+    public SuperTrendIndicator(final BarSeries series, int length, final Integer multiplier) {
+        super(series);
+        ATRIndicator atrIndicator = new ATRIndicator(series, length);
+        this.superTrendUpperBandIndicator = new SuperTrendUpperBandIndicator(series, atrIndicator, multiplier);
+        this.superTrendLowerBandIndicator = new SuperTrendLowerBandIndicator(series, atrIndicator, multiplier);
+    }
+
+    public SuperTrendIndicator(final BarSeries series) {
+        this(series, 10, 3);
+    }
+
+    @Override
+    protected Num calculate(int i) {
+        Num value = ZERO;
+
+        if (i == 0) {
+            return value;
+        }
+        Bar bar = getBarSeries().getBar(i);
+
+        if (this.getValue(i - 1).isEqual(this.superTrendUpperBandIndicator.getValue(i - 1))
+                && bar.getClosePrice().isLessThanOrEqual(this.superTrendUpperBandIndicator.getValue(i))) {
+            value = this.superTrendUpperBandIndicator.getValue(i);
+        }
+
+        if (this.getValue(i - 1).isEqual(this.superTrendUpperBandIndicator.getValue(i - 1))
+                && bar.getClosePrice().isGreaterThan(this.superTrendUpperBandIndicator.getValue(i))) {
+            value = this.superTrendLowerBandIndicator.getValue(i);
+        } else if (this.getValue(i - 1).isEqual(this.superTrendLowerBandIndicator.getValue(i - 1))
+                && bar.getClosePrice().isGreaterThanOrEqual(this.superTrendLowerBandIndicator.getValue(i))) {
+            value = this.superTrendLowerBandIndicator.getValue(i);
+        } else if (this.getValue(i - 1).isEqual(this.superTrendLowerBandIndicator.getValue(i - 1))
+                && bar.getClosePrice().isLessThan(this.superTrendUpperBandIndicator.getValue(i))) {
+            value = this.superTrendUpperBandIndicator.getValue(i);
+        }
```

Comment:
There are 3 issues.

1. Line 59 should use isLessThan instead of isLessThanOrEqual
2. Line 67 should use isGreaterThan instead of isGreaterThanOrEqual 
3. Line 70 should use superTrendLowerBandIndicator instead of superTrendUpperBandIndicator

This is the correct conditionals
```
if (this.getValue(i - 1).isEqual(this.superTrendUpperBandIndicator.getValue(i - 1))
        && bar.getClosePrice().isLessThan(this.superTrendUpperBandIndicator.getValue(i))) {
    value = this.superTrendUpperBandIndicator.getValue(i);
} else if (this.getValue(i - 1).isEqual(this.superTrendUpperBandIndicator.getValue(i - 1))
        && bar.getClosePrice().isGreaterThan(this.superTrendUpperBandIndicator.getValue(i))) {
    value = this.superTrendLowerBandIndicator.getValue(i);
} else if (this.getValue(i - 1).isEqual(this.superTrendLowerBandIndicator.getValue(i - 1))
        && bar.getClosePrice().isGreaterThan(this.superTrendLowerBandIndicator.getValue(i))) {
    value = this.superTrendLowerBandIndicator.getValue(i);
} else if (this.getValue(i - 1).isEqual(this.superTrendLowerBandIndicator.getValue(i - 1))
        && bar.getClosePrice().isLessThan(this.superTrendLowerBandIndicator.getValue(i))) {
    value = this.superTrendUpperBandIndicator.getValue(i);
}
```

Verified with this reference - https://medium.com/codex/step-by-step-implementation-of-the-supertrend-indicator-in-python-656aa678c111

Reply:
I can change the first 2 conditions. Can you check the third condition again? To me it seems correct.

Reply:
![Screenshot 2023-02-24 at 9 44 40 AM](https://user-images.githubusercontent.com/5506862/221090622-c62b909c-316e-4bbb-9b54-a2b545fe6a36.png)
Please check the lines circled in red in the attached screenshot. 
Also when we do not change third point mentioned above, I'm getting the same issue as earlier